pkg/skaffold/deploy/deploy.go

```diff
@@ -21,9 +21,18 @@ import (
 	"fmt"
 	"io"
 
+	"k8s.io/apimachinery/pkg/runtime"
+
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
 )
 
+// DeployResult is a holder struct encapsulating the deployed k8s objects
+type DeployResult struct {
+	obj       *runtime.Object
+	namespace string
+	deployer  string
```

Comment:
We can infer this from the config. I'm OK with adding a `Name() string` requirement to the deploy interface

Reply:
Agreed as per my comment above